executor/executor_test.h

```diff
@@ -5,11 +5,22 @@
 #include <sys/mman.h>
 #include <unistd.h>
 
+#ifdef __unix__
+#include <sys/prctl.h>
+#endif
+
 static uint64 kernel_text_start = 0xc0dec0dec0000000;
 static uint64 kernel_text_mask = 0xffffff;
 
 static void os_init(int argc, char** argv, void* data, size_t data_size)
 {
+#ifdef __unix__
+	prctl(PR_SET_PDEATHSIG, SIGKILL, 0, 0, 0);
+	// There's a risk that the parent has exited before we get to call prctl().
+	// In that case, let's assume that the child must have been reassigned to PID=1.
+	if (getppid() == 1)
```

Comment:
Don't we need this for all other subprocesses (sandbox, test)?
In some cases we already do PR_SET_PDEATHSIG, but don't have the ppid check.

Reply:
I think the question is whether we can be sure that the child process eventually exits. Otherwise (apart from TestOS executors sed in our Go tests) I think we don't care that much if it happens immediately or some ms later?

I wonder if we are going to fail here if the parent has exited -- we call it at the beginning of the loop() function.
https://github.com/google/syzkaller/blob/c6d33a012816b8e651cfc2284a27e9659832143b/executor/executor.cc#L759-L763

Reply:
> I think the question is whether we can be sure that the child process eventually exits. Otherwise (apart from TestOS executors sed in our Go tests) I think we don't care that much if it happens immediately or some ms later?

Yes, in tests to prevent the same remove dir failures.

> I wonder if we are going to fail here if the parent has exited -- we call it at the beginning of the loop() function.

Probably.
But it may be too late for tests.